src/adcm_client/util/search.py

```diff
@@ -72,4 +75,15 @@ def search(data, **attrs):
 
     search returns None if no result found
     """
-    return filter(lambda x: attrs.items() <= x.items(), data)
+
+    def filter_data(x):
+        new_attrs = {}
+        new_data = {}
+        for key, value in x.items():
+            if not isinstance(value, (Mapping, Iterable)):
+                new_data[key] = value
+                if key in attrs:
+                    new_attrs[key] = attrs[key]
+        return new_attrs.items() <= new_data.items()
+
+    return [item for item in data if filter_data(item)]
```

Comment:
It is prefered to return iterator with yield here because of search_one which is uses the fact of second element in search result as an exception.

Reply:
fixed